### ISSUE TITLE
What is param_grouping.py and what is the recommended way for applying weight decay

### ISSUE THREAD

**Issue by MarcelLieb** [state-spaces/mamba#798]
The implementation of Mamba-2 mentions a file named param_grouping.py, but it cannot be found in this repository.

https://github.com/state-spaces/mamba/blob/0b5a3e04e2626f5d0caf3cb669a7f7f70ab502bb/mamba_ssm/modules/mamba2.py#L128-L130
https://github.com/state-spaces/mamba/blob/0b5a3e04e2626f5d0caf3cb669a7f7f70ab502bb/mamba_ssm/modules/mamba2_simple.py#L102-L104

There is also a `_no_weight_decay` attribute that is being set, which is never mentioned anywhere and is ignored by PyTorch by default.

https://github.com/state-spaces/mamba/blob/0b5a3e04e2626f5d0caf3cb669a7f7f70ab502bb/mamba_ssm/modules/mamba2.py#L135-L140
https://github.com/state-spaces/mamba/blob/0b5a3e04e2626f5d0caf3cb669a7f7f70ab502bb/mamba_ssm/modules/mamba2_simple.py#L110-L116
https://github.com/state-spaces/mamba/blob/0b5a3e04e2626f5d0caf3cb669a7f7f70ab502bb/mamba_ssm/modules/mamba_simple.py#L110-L115

I was able to find [param_grouping.py](https://github.com/Dao-AILab/flash-attention/blob/5059fd53e602bcc00336bb5cc8a85e50940485cb/training/src/optim/param_grouping.py), but this leaves the question:

**Is it strongly recommended that weight decay is disabled for `A_log` and `D`?**
**If so, why isn't this mentioned anywhere, and what issues arise when weight decay is applied to `A_log` and `D`?**
